src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -918,6 +934,46 @@ map_injective f.injective u v
 
 end path
 
+/-! ## Deleting edges -/
+
+namespace walk
+variables {G}
+
+/-- Given a walk that avoids a set of edges, produce a walk in the graph
+with those edges deleted. -/
+def to_delete_edges (s : set (sym2 V)) :
+  Π {v w : V} (p : G.walk v w) (hp : ∀ e, e ∈ p.edges → ¬ e ∈ s), (G.delete_edges s).walk v w
+| _ _ nil _ := nil
+| _ _ (cons' u v w h p) hp := cons' u v w (by simpa [h] using hp ⟦(u, v)⟧)
+  (to_delete_edges p (λ e he, hp e (by simp only [he, edges_cons, list.mem_cons_iff, or_true])))
+
+@[simp] lemma to_delete_edges_nil {s : set (sym2 V)} {v : V} :
```

Comment:
Can these lemmas be given by `simps` on the def instead?

Reply:
I think `simps` only works when the value is a structure (that's at least what the error indicates). These `simp` lemmas are a bit silly to have to re-state, since they're just the equation lemmas for the definition.

Reply:
I see - what about tagging it by `simp` instead? If I remember correctly, that means it tags the equation lemmas as simp, which is what you want here

Reply:
Thanks, that worked. (I wonder if it plays well with `squeeze_simp` and `simp?` -- I'm willing to find out later for this.)